### PR TITLE
[K32W1] Update build to use SDK mbedtls and update openthread commit

### PR DESCRIPTION
**Pull request by andrei-menzopol**
Update K32W1 build to use mbedtls from SDK package
Update openthread commit to latest

### CODE REVIEW

**Review by marius-alex-tache — APPROVED** [review]

**Comment by andrei-menzopol** [comment]
Hello @jwhui, could you please review and merge this? It will unblock the update to latest openthread commit. 
It has enough reviews from our side.
Thank you

**Review by jwhui — APPROVED** [review]

**Pull request merged**
